### ISSUE TITLE
Fusion client is trying to reconnect to a dead publisher for too long

### ISSUE THREAD

**Issue by alexyakunin** [servicetitan/Stl.Fusion#263]
The log part indicating this:

```
    Line 2:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
    Line 14:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-do-jCITwOmPKrjyx&clientId=R-da52z70rnU90aPED...
    Line 21:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-lYvW9eC0qQ7tE3bG&clientId=R-da52z70rnU90aPED...
    Line 23:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
    Line 35:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-do-jCITwOmPKrjyx&clientId=R-da52z70rnU90aPED...
    Line 42:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-lYvW9eC0qQ7tE3bG&clientId=R-da52z70rnU90aPED...
    Line 49:       R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui_-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
```

Exception:
```
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui_-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-do-jCITwOmPKrjyx&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-lYvW9eC0qQ7tE3bG&clientId=R-da52z70rnU90aPED...
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui_-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-do-jCITwOmPKrjyx&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-lYvW9eC0qQ7tE3bG&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
info: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: connecting to ws://localhost:9000/fusion/ws?publisherId=P-vdFui_-KP5JB3VFI&clientId=R-da52z70rnU90aPED...
fail: Stl.Fusion.Client.WebSocketChannelProvider[0]
      R-da52z70rnU90aPED: error
      System.Net.WebSockets.WebSocketException (0x80004005): The server returned status code '400' when status code '101' was expected.
         at System.Net.WebSockets.WebSocketHandle.ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         at Stl.Fusion.Client.WebSocketChannelProvider.CreateChannel(Symbol publisherId, CancellationToken cancellationToken)
```

**Comment by alexyakunin** [comment]
To be fixed. For now, a couple comments:
- Attempts to reconnect to a publisher are normal. Though it makes sense to add a test ensuring they'll stop once every replica from this publisher is dead.
- But in this specific case it makes sense to stop immediately once it's clear the publisher is dead - i.e. the response from a server indicates there is no publisher with the specified ID. Publisher IDs aren't supposed to be fixed, so if it's gone - almost certainly it's gone forever, which means every alive replica from this publisher has to be renewed via HTTP, not via WebSocket.

**Comment by alexyakunin** [comment]
Fixed in v3.9.1.
